presto-tests/src/test/java/com/facebook/presto/server/TestClusterStatsResource.java

```diff
@@ -101,15 +111,23 @@ public void testGetClusterStats()
         assertEquals(clusterStats.getQueuedQueries(), 0);
     }
 
-    private ClusterStatsResource.ClusterStats getClusterStats(boolean followRedirects)
+    private ClusterStatsResource.ClusterStats getClusterStats(boolean followRedirects, boolean useThriftEncoding)
     {
-        Request request = prepareGet()
+        Request.Builder builder = prepareGet()
                 .setHeader(PRESTO_USER, "user")
                 .setUri(uriBuilderFrom(server.getBaseUrl().resolve("/v1/cluster")).build())
-                .setFollowRedirects(followRedirects)
-                .build();
-
-        return client.execute(request, createJsonResponseHandler(jsonCodec(ClusterStatsResource.ClusterStats.class)));
+                .setFollowRedirects(followRedirects);
+        if (useThriftEncoding) {
+            Request request = builder
+                    .setHeader("Accept", ThriftRequestUtils.APPLICATION_THRIFT_COMPACT)
```

Comment:
you can use `ThriftRequestUtils.prepareThriftGet` which gives you a builder.
Also this just tests for one protocol and misses out on `APPLICATION_THRIFT_BINARY` and `APPLICATION_THRIFT_FB_COMPACT` . You can create enhance the data provider to add these types as well.